### PR TITLE
CI: Enable caching

### PR DESCRIPTION
**Pull request by Smit-create**
Related issue: #21

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -9,27 +9,59 @@ on:
...
           auto-activate-base: false
 
+      - name: Set cache date
+        run: echo "DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+      - uses: actions/cache@v2
+        with:
+          path: ${{ matrix.prefix }}
+          key: ${{ matrix.label }}-conda-${{ hashFiles('ci/environment.yml') }}-${{ env.DATE }}-${{ env.CACHE_NUMBER }}
+        id: cache
```
Comment (Smit-create) [inline .github/workflows/CI.yml:59]: @certik  Here, the cache will be used for 24 hours. More clearly, the cache of each day will be updated on the first run and can be used for further runs on the same day.

Comment (Smit-create) [reply .github/workflows/CI.yml:59]: or may be we can change to all-time cache with no date:
```
key:
          ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/environment.yml') }}
```

**Comment by certik** [comment]
I think that works, thanks! Do you know what happens when we change the Conda dependencies, say in the `environment.yml` file? Will it automatically rebuild it, or do we have to trigger it manually somehow to update the cache?

**Comment by Smit-create** [comment]
> Will it automatically rebuild it,

I will automatically update accordingly.

**Comment by certik** [comment]
I essentially reverted this in #39. We'll have to try again on this.

**Pull request merged**